modules/seqexec/server/src/main/scala/seqexec/server/InstrumentControllerSim.scala

```diff
@@ -3,27 +3,24 @@
 
 package seqexec.server
 
-import java.util.concurrent.atomic.AtomicBoolean
-import java.util.concurrent.atomic.AtomicInteger
-
 import cats.Show
-import cats.data.EitherT
-import cats.effect.IO
 import cats.effect.Timer
+import cats.effect.Sync
 import cats.implicits._
 import fs2.Stream
-import seqexec.model.dhs.ImageFileId
-import seqexec.server.SeqexecFailure.SeqexecException
 import gov.aps.jca.TimeoutException
+import java.util.concurrent.atomic.AtomicBoolean
+import java.util.concurrent.atomic.AtomicInteger
 import mouse.all._
 import org.log4s.getLogger
-import scala.concurrent.ExecutionContext
+import seqexec.model.dhs.ImageFileId
+import seqexec.server.SeqexecFailure.SeqexecException
 import seqexec.server.InstrumentSystem.ElapsedTime
 import squants.time.{Seconds, Time}
 
 import scala.annotation.tailrec
 
-class InstrumentControllerSim(name: String, useTimeout: Boolean) {
+class InstrumentControllerSim[F[_]: Sync](name: String, useTimeout: Boolean) {
```

Comment:
In general I prefer a model like this

```scala
// an unconstrained tagless interface
trait InstrumentControllerSim[F[_]] {
  def observe(fileId: ImageFileId, expTime: Time): F[ObserveCommand.Result] 
  ...
}

// and then a factory method that does constrain F
object InstrumentControllerSim {
  def apply[F[_]: Sync](name: String, useTimeout: Boolean): InstrumentControllerSim[F] =
    ...
}
```

over a concrete class with constrained `F`, but what you have here is fine for now. I know you're trying to do things systematically.

Reply:
That seems good but how do you know in the implementation. would you need to put the implementation class inside the `object`?